CityOfZion/neon-wallet
# NEP-6 wallet support

**What current issue(s) from Trello/Github does this address?**
https://github.com/CityOfZion/neon-wallet/issues/421

**What problem does this PR solve?**
All projects should follow the NEP-6 wallet standard so they can share wallets

**How did you solve this problem?**
Some notes:
I leave the keys.json storage file behind, just in case.

When we upgrade the wallets we only have labels and encrypted keys. This means that we can’t fill in the address fields until they log into the accounts.

I first tried to use neon-js which was pretty cool but it throws an exception if you try to export an Account without an address. It wasn’t too much work to do it all custom in the end.

Try to use naming inline with the NEP-6 standard (wallets contain accounts, in settings you add accounts not keys)

**How did you make sure your solution works?**
Automated + manual tests

QA Tests worth doing (any others?)
 * Starting with legacy key file, it should migrate to NEP-6. Delete an account, restart app and make sure that account is still missing.
 * Starting with no wallet, test recovering NEP-6 wallet
 * Starting with no wallet, test recovering legacy wallet
 * Adding accounts after the two above
 * Starting with no wallet, test adding 2 accounts, logging into both works 
 * Recovering NEP-6 / Legacy after already having accounts in the wallet
 * Logging into recovered legacy account
 * Logging into recovered NEP-6 account
 * Deleting all those accounts
 * Recovering same recovery file multiple times (won’t allow key to be in wallet twice)
 * Export everything, delete all accounts, import

**Are there any special changes in the code that we should be aware of?**
No.

**Is there anything else we should know?**
Having wallet accounts without public addresses could pose a challenge if other programs the user is importing into don't support that. There's not really anything we can do though.
